sdk/include/opentelemetry/sdk/logs/processor.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include "opentelemetry/logs/log_record.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+/**
+ * This Log Processor is responsible for conversion of logs to exportable
+ * representation and passing them to exporters.
+ */
+class LogProcessor
+{
+public:
+  virtual ~LogProcessor() = default;
+
+  virtual void OnReceive(std::unique_ptr<opentelemetry::logs::LogRecord> &&record) noexcept = 0;
+
+  virtual void ForceFlush(
```

Comment:
My general comment, while we are at it: what is wrong with just `Flush`, and why name `ForceFlush` is better than `Flush`? There is no `ComfortablyOpen` on file descriptors, neither `QuicklyClose` or alike... in that sense I personally feel that word **Force** in a `Flush` - is a misnomer. And simply `Flush` would be much cooler. [fflush](https://man7.org/linux/man-pages/man3/fflush.3.html) - in Linux API is not `f` for force, but rather `f` for file.

Reply:
I can change it to just Flush() here. 
ForceFlush() was how `traces` did it as well - should I change [processor.h](https://github.com/open-telemetry/opentelemetry-cpp/blob/2973baaf476a92e9d11d55e82f1dad78fefe6de1/sdk/include/opentelemetry/sdk/trace/processor.h) for traces as well?

Reply:
The current one looks good as it aligns with the spec.

Reply:
@reyang - can we change the spec? There is never a POSIX or STL library function or any other API for that matter that'd use an adjective before verb. Example of functions with similar semantics:
- fsync
- fflush
- `ostream::flush`
- (Java) `sync`
- (C#) `NetworkStream.Flush`

Reply:
https://github.com/open-telemetry/opentelemetry-specification/issues/1206

Reply:
Not saying to fix in this PR. But while we are at it - designing / adding a new API, I think it's a good opportunity to revisit the naming to make it more consistent with practically any other existing framework.

Reply:
I'm resolving this. Let's merge your PR. We can discuss my concern in spec.